### DISCUSSION TITLE
milvus query or delete based on data filter

### DISCUSSION THREAD

**Discussion by RaflyLesmana3003** [milvus-io/milvus#27459] [Q&A and General discussion] [answered]
currently i am researching about vector database, i already tried using milvus, but i kinda confuse about how exactly collection filter works, based on this documentation its says that milvus only support filter by primary key https://milvus.io/docs/delete_data.md, is it true that i can use another data field for my filter?

**Comment by xiaofan-luan** [comment, marked as answer]
> currently i am researching about vector database, i already tried using milvus, but i kinda confuse about how exactly collection filter works, based on this documentation its says that milvus only support filter by primary key https://milvus.io/docs/delete_data.md, is it true that i can use another data field for my filter?

Are you talking about filtering or delete?
Yes currently delete can be applied on primary key, but you can query or filtering on any expression support. 
Supported expression: https://milvus.io/docs/boolean.md

**Reply by RaflyLesmana3003** [reply]
thank you for answering, so when i want to delete entities, I need to save the primary key on the other database or using query first to get the primary key based on filtering attribute?

**Reply by xiaofan-luan** [reply]
yes, tht's the limit for now.
We have a delete by expr feature in 2.3.2 release

**Reply by RaflyLesmana3003** [reply]
thank you for that information @xiaofan-luan
